### ISSUE TITLE
Could i2p be fast?

### ISSUE THREAD

**Issue by gongzisun** [PurpleI2P/i2pd#1211]
Had someone tried Retroshare? it is a anonymous software, and use a multi hops for transference, and it is fast. I was curious, if i2p's default brandwith setting is unlimited, could i2p work good as the Retroshare?

If so, i think i2pd should change it, Yes, there are some guys get low brandwith like 2G/3G, but it is not hardcode, people can change it themselves. If it hurt someone, maybe i2p could make another version for high brandwith, people can shoose themselves, i believe the unlimited version will win. With high speed, i2p can get a big success.

**Comment by orignal** [comment]
https://github.com/PurpleI2P/i2pd/wiki/Using-with-Retroshare

**Comment by l-n-s** [comment]
It depends on what do you mean by "fast". Currently, max speed I've got is 250KB/s on torrents. Let's hope it will improve with the new fast NTCP2 transport protocol.

Retroshare works fine with i2pd, give it a try.

**Comment by gongzisun** [comment]
Not how to use Retroshare via i2p, i am saying Retroshare's clearnet mode, is the unlimited brandwith matter the transference? Because the most node in i2p/i2pd use the default setting like 250K/s, but Retroshare's default setting did not limit it in clearnet mode, so in Retroshare i can get 1M/s on average.

So is the setting 250K/s(bandwidth = O) the reason why i2p be slow?

**Comment by l-n-s** [comment]
Keep in mind that in i2p your computer send packets through a chain of 12 "proxy" computers (6 out+6 in) for advanced privacy/anonymity. Retroshare does direct connects AFAIK, so your speed is high, but no anonymity at all.

**Comment by majestrate** [comment]
i2p can be fast but the biggest bottleneck is routers on home connections, this is currently an ongoing topic of research. if there was a way to mark a router as "in a data center" then that would help a lot by giving more hints for tunnel building logic to decide who to build over and who to not build over.

**Comment by gongzisun** [comment]
Retroshare's anonymous tunnel work like i2p, i always download something via my friends but they are not the source, i can still get more than 500K/s.

Yes, i2p's hops are too many, every node may affect the final speed, but if the default setting not changed, whatever i change, it is impossible i can get >250K/s speed, so maybe the default setting should be unlimited.

Right, if the default brandwith is unlimited, speed may not be more fast, maybe you can't get >250K/s because someone set it in low level. But if the dafault keep 250K/s, you will never get more than 250K/s.

**Comment by majestrate** [comment]
i suspect java i2p's limiting may cause excessive message drops. also latency introduced from the internet itself, i2p on default settings does 13 hops for each round trip between two destinations for each message, then combine with the message dropping from java i2p (congestion control) and multiple connections over i2p you get upwards of 30 hops if you are talking to a friend of a friend via retroshare.

**Comment by majestrate** [comment]
and you are only as fast as the slowest hop in ALL the hops.

**Comment by gongzisun** [comment]
@majestrate I had said something like that in zzz.i2p last year, like emule's rewards system, the high brandwith get the priority for connecting the high speed node, but just echelon join it, and he think 250K/s is enough, people get 30K/s is the problem, and the reason is latency.

**Comment by majestrate** [comment]
yeah the latency is probably caused by people participating in routing while being on ASDL or some higher latency connection.

**Comment by gongzisun** [comment]
Yes, latency is a reason, but like i said, if the default setting won't be changed, no matter how low latency in the chain, i will never get >250K/s, even in torrent. If every node's brandwith is unlimited, i will get a possibility of getting high speed.
